xrpl/token.js

```diff
@@ -0,0 +1,101 @@
+const { Command } = require('commander');
+const { hex } = require('./utils');
+const { printInfo, printError } = require('../common');
+
+function tokenSymbolToCurrencyCode(tokenSymbol) {
+    if (tokenSymbol.length <= 3) {
+        return tokenSymbol;
+    }
+
+    const hexString = hex(tokenSymbol).toUpperCase();
+
+    if (hexString.length > 40) {
+        printError(
+            `Token symbol "${tokenSymbol}" too long: hex representation (${hexString.length} chars) exceeds xrpl 40-character limit`,
+        );
+        return;
+    }
+    const paddedHex = hexString + '0'.repeat(40 - hexString.length);
+
+    return paddedHex;
+}
+
+function currencyCodeToTokenSymbol(currencyCode) {
+    if (currencyCode.length <= 3) {
+        return currencyCode;
+    }
+
+    if (currencyCode.length !== 40) {
+        printError(`Invalid currency code: ${currencyCode} must be exactly 40 characters (160-bit hex)`);
+        return;
+    }
+
+    if (!/^[0-9A-Fa-f]+$/.test(currencyCode)) {
+        printError(`Invalid currency code: ${currencyCode} is not a valid hex string`);
+        return;
+    }
+
+    const trimmedHex = currencyCode.replace(/0+$/, '');
+
+    if (trimmedHex === '') {
+        return '';
+    }
```

Comment:
this should be an error (more generally, the trimmed number of ascii chars must be >3)